src/main/resources/application.yaml

```diff
@@ -7,3 +7,12 @@ spring:
   jackson:
     serialization:
       write-dates-as-timestamps: false
+  mail:
```

Comment:
I'd prefer to have this mail server configuration in the `application.yaml` that is not checked into source control, since it's something that will be different for every user and can contain secrets. Also, making changes in here requires a recompile and I want to maintain the ability for users to write a config file and just run it directly from Docker without ever touching the code, gradle, Java, etc.

Reply:
Removing it from the `application.yaml` file we have to instruct the user/developer to put them somewhere else and load it from there. Do you have any suggestion on how to accomplish this?

Reply:
One thing to keep in mind, this configuration as it is (in the yaml file) it is automatically picked up by spring. This config is spring specific.

Reply:
The model I've used all along is that there should be an example of all configuration in `application.example.yaml` so you can copy that file, make the changes you want and run it. And if you aren't a developer you can run it directly from the Docker image and it'll pick up your config file that is outside of the JAR and even outside of the Docker image (by mounting it into the image at runtime). I don't see how this email feature should be any different than that; it's an optional feature so some people are going to want to use it and some aren't. So far I haven't put anything into `src/main/resources/application.yaml` that I'd expect a regular user to ever look at or change.

There's also the added benefit that `application.yaml` in the root is included in `.gitignore` so the chances of accidentally committing your secrets is greatly reduced.

Reply:
Aha, I understand now :) 
I forgot the application.example.yaml file existed :sweat_smile:
Yeah I agree, it makes sense. I will commit in a moment.